src/utils/get-functions.js

```diff
@@ -24,4 +26,29 @@ const getFunctions = async (functionsSrcDir) => {
   return functionsWithProps
 }
 
-module.exports = { getFunctions }
+const getWatchDirs = (functionsSrcDir, functions) => {
+  const localFilesDirs = functions
+    .filter(({ srcFile }) => !srcFile.startsWith(functionsSrcDir) && !srcFile.includes('node_modules'))
```

Comment:
Why do we need this `includes('node_modules')`?

Reply:
When calling `listFunctionsFiles` `zisi` returns all the files required by the function (including ones from `node_modules`).

We only want to watch user code.

We pass it also as a flag to `chokidar` too (in case there are `node_modules` under the functions directory): https://github.com/netlify/cli/blob/a2799075df2b61e1653ead3f764ea09b8c5b8e4d/src/utils/serve-functions.js#L185

so I think it will work without it, but I don't see a reason to pass directories we know we don't want to watch, just for them to be ignored later on.

Reply:
Makes sense! Thanks for clarifying.